python/apps/taiga/src/taiga/stories/comments/api.py

```diff
@@ -102,7 +103,35 @@ async def list_story_comments(
 # update story comments
 ##########################################################
 
-# TODO
+
+@routes.comments.patch(
+    "/projects/{project_id}/stories/{ref}/comments/{comment_id}",
+    name="project.story.comments.update",
+    summary="Update story comment",
+    response_model=CommentSerializer,
+    responses=ERROR_403 | ERROR_422 | ERROR_404,
+)
+async def update_story_comments(
+    request: AuthRequest,
+    form: UpdateCommentValidator,
+    project_id: B64UUID,
+    ref: int,
+    comment_id: B64UUID,
+) -> Comment:
+    """
+    Update an story's comment
+    """
+    story = await get_story_or_404(project_id=project_id, ref=ref)
+    comment = await get_story_comment_or_404(comment_id=comment_id, story=story)
+    await check_permissions(permissions=UPDATE_STORY_COMMENT, user=request.user, obj=comment)
+
+    values = form.dict(exclude_unset=True)
+    return await comments_services.update_comment(
+        story=story,
+        comment=comment,
+        values=values,
+        event_on_update=events.emit_event_when_story_comment_is_updated,
```

Comment:
Why is it sent the callable event from the api and don't let the service to manage what event to launch by itself??

Reply:
Following the new way as delete comment.

Reply:
Ok, it's because a generic class and the context of the comment (story) it's in the api. I undertand it. Cool